src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala

```diff
@@ -1086,8 +1087,42 @@ trait SilverTransforms extends SparkSessionWrapper {
         'organization_id, 'cluster_id, 'isRunning,
         'timestamp, 'state, 'current_num_workers, 'target_num_workers
       )
+      .withColumn("unixTimeMS_state_start", 'timestamp)
+      .withColumn("unixTimeMS_state_end", coalesce( // if state end open, use pipelineSnapTime, will be merged when state end is received
+        lead('timestamp, 1).over(stateUnboundW) - lit(1), // subtract 1 millis
+        lit(untilTime.asUnixTimeMilli)
+      ))
+
+    // Change for PR -934
+    // Get the ClusterID that has been Permenantly_Deleted
+    val clusterBaseDF = clusterBase(auditLogDF)
+    val clustersRemoved = clusterBaseDF
+      .filter('actionName.isin("permanentDelete"))
+      .select('timestamp.alias("deletion_timestamp"),'organization_id, 'cluster_id, 'userEmail.alias("deleted_by"))
+
+    val removedClusterID = clustersRemoved.select("cluster_id","deletion_timestamp").distinct()
+    val clusterEventsBaselineForRemovedCluster = clusterEventsBaseline.join(removedClusterID,Seq("cluster_id"))
 
-    clusterEventsBaseline
+    val window = Window.partitionBy('organization_id, 'cluster_id).orderBy('timestamp.desc)
+    val stateBeforeRemoval = clusterEventsBaselineForRemovedCluster.withColumn("rnk",rank().over(window)).filter('rnk === 1).drop("rnk")
+
+    val stateDuringRemoval = stateBeforeRemoval
```

Comment:
Why is the permanent delete OW_runID and Pipeline_SnapTS the same as the one above it. We should be filling those. Look at all the columns and be sure to define each column appropriately.

![image](https://github.com/databrickslabs/overwatch/assets/10840635/d9198799-60f0-4f64-8923-d2da64cdae26)

Reply:
I think it is showing correct value. Can we discuss the same in our daily call